Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -33,9 +36,9 @@ public class OpenFeatureAPI {
         self._provider = nil
     }
 
-    public func setEvaluationContext(evaluationContext: EvaluationContext) async {
+    public func setEvaluationContext(evaluationContext: EvaluationContext) {
+        getProvider()?.onContextSet(oldContext: self._context, newContext: evaluationContext)
```

Comment:
Any reason why this line was moved up? Setting the global context before starting any side effect in the Provider is important; more info here: [PR link](https://github.com/spotify/openfeature-swift-sdk/pull/19/files#diff-6e3a4f136b63288df5178198c6468514ea94dddd9e188495502016393ab76d02R38)

Reply:
The reasoning was because if we set `self._context = evaluationContext` before calling `onContextSet`, we're then effectively calling it with `oldContext` and `evaluationContext` as the same context. However, maybe the solution should be this?

```
let oldContext = self._context
self._context = evaluationContext
getProvider()?.onContextSet(oldContext: oldContext, newContext: evaluationContext)
```

Reply:
Good point, that was totally a bug introduced in my previous PR. Yes I think the solution you proposed would be best 👍